### DISCUSSION TITLE
How do I validate a validation error response?

### DISCUSSION THREAD

**Discussion by bachikawa** [vitalik/django-ninja#1118] [Q&A] [answered]
My API has a default validation error that is overridden by my own definition. It is defined in a way using `@api.exception_handler` based on the official documentation and returns an error response by `api.create_response()`.

For a normal response (status 200), I know that the response content is validated by using `response={200: MyResponseSchema}`.

But how can I validate the default validation error as well as the normal response?

I put `response={200: MyResponseSchema, 400: MyValidationErrorSchema}` in my code.

In my opinion, the auto-generated openapi may not match the actual API specification, since validation errors are not handled in the View.

My desire is to have the auto-generated openapi and the error response returned by the API match exactly at the code level. I believe this will ensure accurate documentation.

I am a native Japanese speaker, so I am using the power of a translator. Thanks for reading.

**Comment by matez0** [comment]
Hi @bachikawa,
The proposal https://github.com/vitalik/django-ninja/discussions/1253#discussion-7012042 may solve this issue.

**Reply by bachikawa** [reply]
Hello.
Thanks for your comment.
Sorry for the delay in responding due to health reasons.
If you have a way to solve this issue #1118, I would love to see you create a PR.
Thanks.

**Reply by matez0** [reply, marked as answer]
Hi @bachikawa,
Here is a pull request for the https://github.com/vitalik/django-ninja/discussions/1253#discussion-7012042 proposal:
https://github.com/vitalik/django-ninja/pull/1407

**Reply by bachikawa** [reply]
Thank you very much!
I need this and I trust that many others will find this to be good PR #1407 as well!
Thank you!

By the way, how should we handle this discussion at now?
Can I mark your comment as an answer?

**Reply by matez0** [reply]
Welcome.
Similar could be done with the FastAPI as well.
Actually, the PR would be the real answer, but of course, you can mark the idea as well.

**Reply by bachikawa** [reply]
Thank you!
